automated-testing/ui-testing/readme.md

```diff
@@ -0,0 +1,64 @@
+# UI Testing
+
+This section is primarily geared towards web-based UIs, but the guidance is similar for mobile and OS based applications.  
+
+## Goals
+
+UI testing provides the ability to ensure that users have a consistent visual user experience across a variety of means of access and that the user interaction is consistent with the function requirements.
+
+### UI tests have several goals
+
+- Ensure the UI appearance and interaction satisfy the functional and non-functional requirements
+- Detect changes in the UI both across devices and delivery platforms and between code changes
+- Provide confidence to designers and developers the user experience is consistent
+- Support fast code evolution and refactoring while reducing the risk of regressions
+
+## Evidence and Measures
+
+Integrating UI Tests in to your CI/CD is necessary but more challenging than unit tests.  The increased challenge is that UI tests either need to run headlessly with something like [Puppeteer](https://github.com/puppeteer/puppeteer) or there needs to be more extensive orchestration with Azure DevOps or Github that would handle the full testing integration for you like [BrowserStack](https://www.browserstack.com/automate/azure)
+
+Integrations like `BrowserStack` are nice since they provide Azure DevOps reports as part of the test run.
```

Comment:
I was attempting to not make specific recommendations based on the CI/CD system, though I did mention BrowserStack has AzDo integration, it has GitHub support through TravisCI as well.  For the most part, I think each plugin/framework I mentioned can be used in any CI/CD system, which is why I didn't make the distinction.   Given that, I'm not entirely sure how to break it up since I think each testing framework works pretty much anywhere.  Happy to hear your suggestion on breaking it up.